worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/FileWriter.java

```diff
@@ -440,4 +449,10 @@ public void notifyNonCriticalError(String mountPoint, DiskStatus diskStatus) {}
   public PartitionType getPartitionType() {
     return partitionType;
   }
+
+  public void setShuffleKeyAndStorageManagerWhenGracefulShutdownIsEnabled(
```

Comment:
IMO better to rename to setShuffleKeyAndStorageManager, or separate into setShuffleKey and setStorageManager as before, if you don't want to pass in the constructor.